src/cephadm/cephadm

```diff
@@ -977,11 +978,16 @@ def get_parm(option):
     if not option:
         return dict()
 
+    global cached_stdin
     if option == '-':
-        try:
-            j = injected_stdin  # type: ignore
-        except NameError:
-            j = sys.stdin.read()
+        if cached_stdin is not None:
+            j = cached_stdin
+        else:
+            try:
+                j = injected_stdin  # type: ignore
```

Comment:
This construct has the big architectural downside of making sure, we'll never be able to replace this single-big-cephadm script. I'd which there would be a sane alternative.